mne/io/fiff/raw.py

```diff
@@ -115,13 +115,14 @@ def __init__(self, fname, allow_maxshield=False, preload=False,
             first_samps = list()
             last_samps = list()
             for r in raws:
-                first_samps = np.r_[first_samps, r.first_samp]
-                last_samps = np.r_[last_samps, r.last_samp]
                 self.annotations = _combine_annotations((self.annotations,
                                                          r.annotations),
                                                         last_samps,
                                                         first_samps,
-                                                        r.info['sfreq'])
+                                                        r.info['sfreq'],
+                                                        self.info['meas_date'])
+                first_samps = np.r_[first_samps, r.first_samp]
+                last_samps = np.r_[last_samps, r.last_samp]
```

Comment:
Either way is fine by me. Will this only be used for very long FIF files (>2GB)? If so it might not be too hard to test with a small `split_size` in `raw.save` followed by `read_raw_fif` and comparison of `annotations`.